### PR TITLE
Python 3: make sure repo path is str type

### PR DESCRIPTION
**Pull request by ziransun**
This CL is to fix a TypeError for edgechromium run with Python 3.6

### CODE REVIEW

**Comment by jgraham** [comment]
I get mildly nervous when things are required to be a native str; can you comment in the code to explain what's going on?

File: tools/wptrunner/wptrunner/vcs.py [hunk 1]
```diff
@@ -54,7 +56,7 @@ def bind_to_repo(vcs_func, repo, log_error=True):
 
 def is_git_root(path, log_error=True):
     try:
-        rv = git("rev-parse", "--show-cdup", repo=path, log_error=log_error)
+        rv = git("rev-parse", "--show-cdup", repo=isomorphic_decode(path), log_error=log_error)
```
Comment (stephenmcgruer) [inline tools/wptrunner/wptrunner/vcs.py:59]: Could we workaround this inside `git` instead (which is `def inner(...)` in `tools/wptrunner/wptrunner/vcs.py`), where `repo` is used as the value for the `cwd` parameter in popen? That's still a workaround for our types being messed up in this code, but it at least contains the workaround to somewhere where we can start working backwards to fix it.

Comment (stephenmcgruer) [reply tools/wptrunner/wptrunner/vcs.py:59]: Incidentally, `cwd` is meant to be able to be a byte string, but that was only fixed in 3.8 on Windows >_<.

```
If cwd is not None, the function changes the working directory to cwd before executing the child. cwd can be a string, bytes or path-like object. In particular, the function looks for executable (or for the first item in args) relative to cwd if the executable path is a relative path.

Changed in version 3.6: cwd parameter accepts a path-like object on POSIX.

Changed in version 3.7: cwd parameter accepts a path-like object on Windows.

Changed in version 3.8: cwd parameter accepts a bytes object on Windows.
```

Comment (ziransun) [reply tools/wptrunner/wptrunner/vcs.py]: Updated. Thank you!

File: tools/wptrunner/wptrunner/vcs.py [hunk 2]
```diff
@@ -23,7 +25,16 @@ def inner(command, *args, **kwargs):
...
-            proc_kwargs["cwd"] = repo
+            # Make sure repo is str type to work in different sub-versions of Python 3
+            # https://docs.python.org/3/library/subprocess.html
+            # "If cwd is not None, the function changes the working directory to cwd before
+            # executing the child. cwd can be a string, bytes or path-like object. In particular,
+            # the function looks for executable (or for the first item in args) relative to cwd
+            # if the executable path is a relative path.
+            # Changed in version 3.6: cwd parameter accepts a path-like object on POSIX.
+            # Changed in version 3.7: cwd parameter accepts a path-like object on Windows.
+            # Changed in version 3.8: cwd parameter accepts a bytes object on Windows."
```
Comment (stephenmcgruer) [inline tools/wptrunner/wptrunner/vcs.py:36]: This is a beautifully in-depth comment, but I think maybe a little *too* in-depth 😆 . How about:

```suggestion
            # Make sure `cwd` is str type to work in different sub-versions of Python 3.
            # Before 3.8, bytes were not accepted on Windows for `cwd`.
```

Comment (ziransun) [reply tools/wptrunner/wptrunner/vcs.py]: updated :)

**Review by stephenmcgruer — APPROVED** [review]
LGTM. I'll give @jgraham 24 hours to raise any concerns otherwise I will merge tomorrow.

**Pull request merged**